velox/exec/HashPartitionFunction.cpp

```diff
@@ -84,4 +84,59 @@ void HashPartitionFunction::partition(
     }
   }
 }
+
+std::unique_ptr<core::PartitionFunction> HashPartitionFunctionSpec::create(
+    int numPartitions) const {
+  return std::make_unique<exec::HashPartitionFunction>(
+      numPartitions, inputType_, keyChannels_, constValues_);
+}
+
+std::string HashPartitionFunctionSpec::toString() const {
+  std::vector<std::string> constValueStrs;
+  constValueStrs.reserve(constValues_.size());
+  for (const auto& value : constValues_) {
+    constValueStrs.emplace_back(value->toString());
+  }
+  return constValueStrs.empty()
+      ? fmt::format("HASH(keyChannels:{})", folly::join(", ", keyChannels_))
```

Comment:
Do we have a test case for toString() output when there are constant values?

Reply:
Yes